vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/FileManager.java

```diff
@@ -0,0 +1,108 @@
+package edu.asu.diging.vspace.core.services.impl;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.apache.tika.Tika;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.io.ByteArrayResource;
+import org.springframework.core.io.Resource;
+import org.springframework.stereotype.Service;
+
+
+import edu.asu.diging.vspace.core.data.FileRepository;
+import edu.asu.diging.vspace.core.exception.FileStorageException;
+import edu.asu.diging.vspace.core.factory.IFileFactory;
+import edu.asu.diging.vspace.core.file.IStorageEngine;
+import edu.asu.diging.vspace.core.model.IVSFile;
+import edu.asu.diging.vspace.core.model.impl.VSFile;
+import edu.asu.diging.vspace.core.services.IFileManager;
+
+@Service
+public class FileManager implements IFileManager {
+    
+    @Autowired
+    private IStorageEngine storageEngine;
+    
+    @Autowired
+    private IFileFactory fileFactory;
+    
+    @Autowired
+    private FileRepository fileRepo;
+    
+    @Value("${file_uploads_directory}")
+    private String fileUploadDir;
```

Comment:
that should not be needed here. either the storage engine is the one deciding where to store something,  or this class but not both.